libbeat/outputs/kafka/config.go

```diff
@@ -14,34 +14,58 @@ type kafkaConfig struct {
 	TLS             *outputs.TLSConfig `config:"tls"`
 	Timeout         time.Duration      `config:"timeout"             validate:"min=1"`
 	Worker          int                `config:"worker"              validate:"min=1"`
+	Metadata        metaConfig         `config:"metadata"`
 	UseType         bool               `config:"use_type"`
 	Topic           string             `config:"topic"`
 	KeepAlive       time.Duration      `config:"keep_alive"          validate:"min=0"`
 	MaxMessageBytes *int               `config:"max_message_bytes"   validate:"min=1"`
 	RequiredACKs    *int               `config:"required_acks"       validate:"min=-1"`
 	BrokerTimeout   time.Duration      `config:"broker_timeout"      validate:"min=1"`
 	Compression     string             `config:"compression"`
+	Version         string             `config:"version"`
 	MaxRetries      int                `config:"max_retries"         validate:"min=-1,nonzero"`
 	ClientID        string             `config:"client_id"`
 	ChanBufferSize  int                `config:"channel_buffer_size" validate:"min=1"`
+	Username        string             `config:"username"`
+	Password        string             `config:"password"`
+}
+
+type metaConfig struct {
+	Retry       metaRetryConfig `config:"retry"`
+	RefreshFreq time.Duration   `config:"refresh_frequency" validate:"min=0"`
+}
+
+type metaRetryConfig struct {
+	Max     int           `config:"max"     validate:"min=0"`
+	Backoff time.Duration `config:"backoff" validate:"min=0"`
 }
 
 var (
 	defaultConfig = kafkaConfig{
-		Hosts:           nil,
-		TLS:             nil,
-		Timeout:         30 * time.Second,
-		Worker:          1,
+		Hosts:   nil,
+		TLS:     nil,
+		Timeout: 30 * time.Second,
+		Worker:  1,
+		Metadata: metaConfig{
```

Comment:
sarama lib supports kafka 0.8 - 0.10 . Settings should be valid for all kafka versions. Lib might internally ignore settings if they do not apply to the version being used. AFAIK metadata updates and publisher settings should have same effect for kafka versions.